### PR TITLE
IconTextField: make mouse listener affect the whole area

### PR DESCRIPTION
**Pull request by Trevor159**
I was looking into adding doubling clicking the search bar in the hiscore panel to lookup the local player like OSB used to have. I never realized we already had this because right now it only works if you double click the picture section of the search bar which is a low percent of the total area. This lets you also double click the textbox part of the search bar.

### CODE REVIEW

**Comment by Trevor159** [comment]
In case it wasn't obvious here is a picture



The left of the red line is what you can double click currently, with this PR you can click the whole box.

**Comment by Adam-** [comment]
Shouldn't there be a removeMouseListener too?

**Comment by Trevor159** [comment]
> Shouldn't there be a removeMouseListener too?

I can make one if wanted. It was not used at all in the client currently for these classes so I didn't.

**Comment by deathbeam** [comment]
I agree, you should add also remove part, even if it is unused

**Comment by Adam-** [comment]
Why is the icon text field mouse listener added to the panel and the flat text field not?

**Comment by Trevor159** [comment]
> Why is the icon text field mouse listener added to the panel and the flat text field not?

I am not a swing expert so this could be wrong but the reason I did it is because in FlatTextField the JTextField covers the whole panel so adding the mouse listener to the panel would be redundant.

**Comment by Adam-** [comment]
Why is it adding the listener to the jtextfield in flattextfield then if the panel already covers it?

**Comment by deathbeam** [comment]
So, as far as I remember:
1. This functionality was originally intentional
2. Panel is not covering anything, you *need* to add listeners to child components in Swing if you want some hover effect or w/e, that is just how it works (unless it's just label, that one is pass-through I think)

File: runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java [hunk 1]
```diff
@@ -169,6 +169,20 @@ public void changedUpdate(DocumentEvent e)
 		add(clearButton, BorderLayout.EAST);
 	}
 
+	@Override
+	public void addMouseListener(MouseListener mouseListener)
+	{
+		super.addMouseListener(mouseListener);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java]: should this really apply to whole icon text field? and not just text field and icon?

File: runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java [hunk 2]
```diff
@@ -169,6 +169,20 @@ public void changedUpdate(DocumentEvent e)
...
+		textField.addMouseListener(mouseListener);
+	}
+
+	@Override
+	public void removeMouseListener(MouseListener mouseListener)
+	{
+		super.removeMouseListener(mouseListener);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java]: see my previous comment

**Comment by raiyni** [comment]
Wouldn't this be fairly confusing behavior? Go in expecting to double click to highlight whole like (like every other text field that exists) and instead it does a lookup on your rsn.

**Comment by deathbeam** [comment]
Maybe this could check if the text field is empty then?

**Comment by Nightfirecat** [comment]
A suggestion from #8968 which I thought might help this discussion--maybe a right-click menu option would work here?

**Comment by Trevor159** [comment]
I don't know why I ever cared enough to make this PR.

**Pull request closed without merging**